app/src/components/App/App.js

```diff
@@ -1,11 +1,196 @@
+  const [signInState, setSignInState] = React.useState(true);
+  const [signUpState, setSignUpState] = React.useState(true);
   const [isMobileNavbarOpen, setIsMobileNavbarOpen] = React.useState(false);
+  const [isSuccessFormOpened, setIsSuccessformOpened] = React.useState(false);
+  const [isRegistered, setIsRegistered] = React.useState(false);
+  const [isLoggedIn, setIsLoggedIn] = React.useState(false);
+  const [currentUser, setCurrentUser] = React.useState({});
+  const [articles, setArticles] = React.useState([]);
+  const [maxArticlesRows, setMaxArticlesRows] = React.useState(1);
+  const [searchWorld, setSearchWorld] = React.useState("");
+  const [preloaderOpened, setPreloaderOpened] = React.useState(false);
+  const [savedNews, setSavedNews] = React.useState([]);
+  const [signUpError, setSignUpError] = React.useState("");
+  const [token, setToken] = React.useState(localStorage.getItem("token"));
+  const [errors, setErrors] = React.useState({
+    email: "",
+    password: "",
+    username: "",
+  });
+  const [data, setData] = React.useState({
+    email: "",
+    password: "",
+    username: "",
+  });
+
+  React.useEffect(() => {
+    if (token) {
+      mainApi
+        .getSavedArticles(token)
+        .then((articles) => {
+          if (articles) {
+            setSavedNews(articles.data);
+          }
+        })
+        .catch((err) => console.log(err));
+    }
+  }, [token, savedNews]);
+
+  React.useEffect(() => {
+    if (token) {
+      mainApi
+        .checkToken(token)
+        .then((res) => {
+          setCurrentUser(res.data);
+          setIsLoggedIn(true);
+        })
+        .then(() => {
+          navigate("/saved-news");
+        })
+        .catch((err) => console.log(err));
+    } else {
+      setIsLoggedIn(false);
+    }
+  }, [token]);
+
+  const closeAllPopups = (e) => {
+    e.preventDefault();
+    setIsSignInformOpened(false);
+    setIsSignUpformOpened(false);
+    setIsSuccessformOpened(false);
+    setData({
+      email: "",
+      password: "",
+      username: "",
+    });
+    setErrors({
+      email: "",
+      password: "",
+      username: "",
+    });
+    setSignInState(true);
+    setSignUpState(true);
+    setSignUpError("");
+  };
+
+  const handleDelete = (token, id) => {
+    return mainApi.deleteArticle(token, id);
+  };
+
+  const handleLogoutClick = (e) => {
+    e.preventDefault();
+    setIsLoggedIn(false);
+    setToken("");
+
+    localStorage.removeItem("token");
+  };
+
+  const handleSignUp = (e) => {
+    e.preventDefault();
+    mainApi
+      .register(data.email, data.password, data.username)
+      .then((data) => {
+        if (!data.message) {
+          closeAllPopups(e);
+          return data;
+        } else {
+          setSignUpError(data.message);
+        }
+      })
+      .then((data) => {
+        if (data) {
+          setIsRegistered(true);
+          setIsSuccessformOpened(true);
+        }
+      });
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**